### ISSUE TITLE
[prometheus-kube-stack] Prometheus ingress is missing pathType 

### ISSUE THREAD

**Issue by toreshonda** [prometheus-community/helm-charts#973]
**Describe the bug**
Trying to install prometheus with enable ingress for prometheus server

**Version of Helm and Kubernetes**:

Helm Version:

```console
$ helm version
version.BuildInfo{Version:"v3.5.4", GitCommit:"1b5edb69df3d3a08df77c9902dc17af864ff05d1", GitTreeState:"clean", GoVersion:"go1.15.11"}
```

Kubernetes Version:

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-12T14:18:45Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-12T14:12:29Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}

```

**Which chart**: prometheus

**Which version of the chart**: 15.4.6

**What happened**: 
If enable Prometheus ingress in true get error:
Error: Ingress.extensions "prom-prometheus" is invalid: spec.rules[0].http.paths[0].pathType: Required value: pathType must be specified
generated template for prometheus ingress:

apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kubernetes.io/tls-acme: "true"
  name: prom-prometheus
  namespace: monitoring
  labels:
    app: core-prometheus

    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/instance: core
    app.kubernetes.io/version: "15.4.6"
    app.kubernetes.io/part-of: core
    chart: kube-prometheus-stack-15.4.6
    release: "core"
    heritage: "Helm"
spec:
  rules:
    - host: dev-prometheus.nces.by
      http:
        paths:
          - path: /
            backend:
              service:
                name: prom-prometheus
                port:
                  number: 9090
  tls:
    - hosts:
      - dev-prometheus.nces.by
      secretName: nces-tls

**What you expected to happen**:
add pathType: Prefix or ImplementationSpecific  to 
[https://github.com/prometheus-community/helm-charts/blob/a5279189e8011136d3aaeaa06f36c3f9b3f893ce/charts/prometheus/templates/server/ingress.yaml#L37](url)

**How to reproduce it** : Simply try to install chart with enabled prometheus ingress in values.yaml
**Changed values of values.yaml** (only put values which differ from the defaults):

values.yaml

```console
prometheus:
...
  ingress:
    enabled: true
    annotations:
      kubernetes.io/ingress.class: nginx
      kubernetes.io/tls-acme: "true"
    labels: {}
    hosts:
      - {{ prometheus_ingress_host }}
    paths:
      - /
    tls:
      - secretName: {{ ingress_tls_name }}
        hosts:
          - {{ prometheus_ingress_host }}
```

**The helm command that you execute and failing/misfunctioning**:

For example:

```console
helm install my-release prometheus-community/name-of-chart --version version --values values.yaml
```

**Helm values set after installation/upgrade:**

```console
helm get values my-release
```

**Anything else we need to know**:

**Comment by idlab-org** [comment]
The same error affects AlertManager:
```
alertmanager:
  ingress:
    enabled: true
    hosts:
       - alertmanager.domain.com
    paths:
      - /
```
Results in `Error: Ingress.extensions "prometheus-stack-kube-prom-alertmanager" is invalid: spec.rules[0].http.paths[0].pathType: Required value: pathType must be specified`

**Comment by DnR-iData** [comment]
:wave:

Adding `pathType: Prefix` resolved the error :

```yaml
alertmanager:
  ingress:
    enabled: true
    hosts:
       - alertmanager.domain.com
    paths:
      - /
    pathType: Prefix
```

And same key/value for prometheus.


```yaml
prometheus:
  ingress:
    enabled: true
    hosts:
       - prometheus.domain.com
    paths:
      - /
    pathType: Prefix
```

Finded information into `values.yaml` https://github.com/prometheus-community/helm-charts/blob/d1525b804b2b49dab984369743cbea8de180c410/charts/kube-prometheus-stack/values.yaml#L231 :

```yaml
    ## For Kubernetes >= 1.18 you should specify the pathType (determines how Ingress paths should be matched)
    ## See https://kubernetes.io/blog/2020/04/02/improvements-to-the-ingress-api-in-kubernetes-1.18/#better-path-matching-with-path-types
    # pathType: ImplementationSpecific
```

DR

**Comment by TheKangaroo** [comment]
I can confirm that @DnR-iData solution solved this issue.
I wonder though, if `pathType: Prefix` should be the default in values?
This line (https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus/ingress.yaml#L38) already ensures to set pathType only on compatible ingress api versions, so I think it would be best to default pathType to Prefix.

**Comment by peterrosell** [comment]
I agree with @TheKangaroo. As the `path` value is set we should also provide the `pathType` and as it's a prefix it should be set to `Prefix` so it's compatible with any ingress controller.

**Comment by irizzant** [comment]
I agree `pathType: Prefix` should be included everywhere in the chart's ingresses

**Comment by irizzant** [comment]
Activity!
